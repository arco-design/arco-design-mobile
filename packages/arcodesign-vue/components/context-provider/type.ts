export interface ContextInstallOptions {
    /**
     * 组件名称前缀
     * @en Component prefix
     * @default "Arco"
     */
    componentPrefix?: string;
}

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
}
