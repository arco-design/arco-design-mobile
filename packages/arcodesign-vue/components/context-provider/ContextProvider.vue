<script lang="ts" setup>
import { provide } from 'vue';
import { GlobalContextParams } from './type';
import { configProviderInjectionKey, defaultContext } from './utils';

const props = withDefaults(defineProps<GlobalContextParams>(), defaultContext);
provide(configProviderInjectionKey, props);
</script>
<template>
    <slot />
</template>
