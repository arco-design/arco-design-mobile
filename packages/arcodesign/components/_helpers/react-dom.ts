import { ReactElement } from 'react';
import * as ReactDOM from 'react-dom';

function isObject(obj: any): obj is { [key: string]: any } {
    return Object.prototype.toString.call(obj) === '[object Object]';
}

export interface RootType {
    render: (element: ReactElement) => void;
    _unmount: () => void;
}
export interface RootTypeReact extends RootType {
    unmount?: () => void;
}
export type CreateRootFnType = (container: Element | DocumentFragment) => {
    render: (element: ReactElement) => void;
    unmount?: () => void;
};

// Cast ReactDOM to a version that might have createRoot and other properties
const typedReactDOM = ReactDOM as typeof ReactDOM & {
    createRoot?: CreateRootFnType;
    render?: (
        element: ReactElement,
        container: Element | DocumentFragment,
        callback?: () => void,
    ) => any;
    unmountComponentAtNode?: (container: Element | DocumentFragment) => boolean;
    __SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED?: {
        usingClientEntryPoint?: boolean;
    };
};

let copyRender: (
    app: ReactElement,
    container: Element | DocumentFragment,
    createRootFunction?: CreateRootFnType,
) => RootType;

const updateUsingClientEntryPoint = (skipWarning?: boolean) => {
    // https://github.com/facebook/react/blob/17806594cc28284fe195f918e8d77de3516848ec/packages/react-dom/npm/client.js#L10
    // Avoid console warning
    const secretInternals = typedReactDOM.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    if (isObject(secretInternals)) {
        secretInternals.usingClientEntryPoint = skipWarning;
    }
};

const createRootFn = typedReactDOM.createRoot;

const getRender =
    (createRootFunction: CreateRootFnType) =>
    (app: ReactElement, container: Element | DocumentFragment) => {
        updateUsingClientEntryPoint(true);
        const root = createRootFunction!(container);
        updateUsingClientEntryPoint(false);

        root.render(app);

        return {
            render: (elementToRender: ReactElement) => {
                root.render(elementToRender);
            },
            _unmount() {
                setTimeout(() => {
                    if (root && typeof root.unmount === 'function') {
                        root.unmount();
                    }
                });
            },
        };
    };

if (createRootFn) {
    // React 18
    copyRender = getRender(createRootFn);
} else if (
    typeof typedReactDOM.render === 'function' &&
    typeof typedReactDOM.unmountComponentAtNode === 'function'
) {
    // React 16/17
    copyRender = function (app: ReactElement, container: Element | DocumentFragment) {
        typedReactDOM.render!(app, container); // Use non-null assertion
        return {
            render: (elementToRender: ReactElement) => {
                typedReactDOM.render!(elementToRender, container); // Use non-null assertion
            },
            _unmount() {
                typedReactDOM.unmountComponentAtNode!(container); // Use non-null assertion
            },
        };
    };
} else {
    copyRender = (
        app: ReactElement,
        container: Element | DocumentFragment,
        createRootFunction?: CreateRootFnType,
    ) => {
        const defaultCb = () => {
            // Fallback if no rendering method is found
            console.error(
                'ArcoDesign: ReactDOM.createRoot() or ReactDOM.render() not found. ' +
                    'This usually means you are using an unsupported version of React, ' +
                    'or ReactDOM is not properly initialized. ' +
                    'ArcoDesign Mobile React requires React 16, 17, 18, or 19.',
            );
            return { render: (_element: ReactElement) => {}, _unmount: () => {} };
        };
        if (createRootFunction) {
            return getRender(createRootFunction)(app, container);
        }
        return defaultCb();
    };
}

export const render = copyRender;
