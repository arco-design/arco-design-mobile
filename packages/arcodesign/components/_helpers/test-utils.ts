import React, { useEffect, useRef } from 'react';
import { Promise as ES6Promise, Promise } from 'es6-promise';

/**
 * React 19 兼容的 act 包装器
 * 用于包装可能触发状态更新的异步操作
 */
export const actWrapper = (callback: () => void | ES6Promise<void>): ES6Promise<void> => {
    return new ES6Promise<void>((resolve, reject) => {
        try {
            // 尝试使用 react-dom/test-utils 的 act
            const testUtils = require('react-dom/test-utils');
            if (testUtils && typeof testUtils.act === 'function') {
                // 对于同步回调
                if (typeof callback === 'function') {
                    try {
                        const result = callback();
                        if (result && typeof result.then === 'function') {
                            testUtils
                                .act(() => result)
                                .then(() => resolve())
                                .catch(reject);
                        } else {
                            testUtils.act(() => {
                                callback();
                            });
                            resolve();
                        }
                    } catch (error) {
                        reject(error);
                    }
                } else {
                    resolve();
                }
            } else {
                const result = callback();
                if (result && typeof result.then === 'function') {
                    result.then(() => resolve()).catch(reject);
                } else {
                    resolve();
                }
            }
        } catch (error) {
            // 如果 act 不可用，直接执行回调
            try {
                const result = callback();
                if (result && typeof result.then === 'function') {
                    result.then(() => resolve()).catch(reject);
                } else {
                    resolve();
                }
            } catch (callbackError) {
                reject(callbackError);
            }
        }
    });
};

/**
 * 清理未处理的异步操作，防止内存泄漏警告
 */
export const cleanupAsync = (): ES6Promise<void> => {
    return new ES6Promise<void>(resolve => {
        setTimeout(() => resolve(), 0);
    });
};

/**
 * 安全的状态更新，检查组件是否已卸载
 */
export const createSafeStateUpdater = (isMountedRef: React.MutableRefObject<boolean>) => {
    return (updater: () => void) => {
        if (isMountedRef.current) {
            updater();
        }
    };
};

/**
 * 用于测试的 useEffect 清理函数
 */
export const useTestCleanup = (cleanupFn: () => void) => {
    const isMountedRef = useRef(true);

    useEffect(() => {
        return () => {
            isMountedRef.current = false;
            cleanupFn();
        };
    }, [cleanupFn]);

    return isMountedRef;
};
