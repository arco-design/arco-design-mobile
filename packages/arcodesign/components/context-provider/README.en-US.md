### Others

# ContextProvider 

Global data control component, used to replace global data.

======

> Props

|Property|Description|Type|DefaultValue|
|----------|-------------|------|------|
|prefixCls|Component classname prefix|string|"arco"|
|system|Manually control the current system, and the incoming value will be used directly after being passed in\. It is applicable when the initial value of the system needs to be specified in the ssr scenario\.|"" \| "pc" \| "android" \| "ios"|""|
|useDarkMode|Whether to use dark mode|boolean|false|
|isDarkMode|Whether it is in dark mode|boolean|false|
|theme|Theme variable\. The css variable will be replaced online after input\. The less variable needs to be set|Record\<string, string\>|-|
|locale|Internationalized language configuration|ILocale|-|
|useRtl|Whether to use rtl|boolean|false|

> ILocale

|Property|Description|Type|
|----------|-------------|------|
|locale|Language Type|string|
|LoadMore|-|\{ loadMoreText: string; loadingText: string; prepareText: string; noDataText: string; failLoadText: string; prepareScrollText: string; prepareClickText: string; \}|
|Picker|-|\{ okText: string; cancelText: string; \}|
|Tag|-|\{ addTag: string; \}|
|Dialog|-|\{ okText: string; cancelText: string; \}|
|SwipeLoad|-|\{ normalText: string; activeText: string; \}|
|PullRefresh|-|\{ loadingText: string; pullingText: string; finishText: string; loosingText: string; \}|
|DropdownMenu|-|\{ select: string; \}|
|Pagination|-|\{ previousPage: string; nextPage: string; \}|
|Image|-|\{ loadError: string; \}|
|ImagePicker|-|\{ loadError: string; \}|
|SearchBar|-|\{ placeholder: string; cancelBtn: string; \}|
|Stepper|-|\{ minusButtonName: string; addButtonName: string; \}|
|Keyboard|-|\{ confirm: string; \}|
|Form|-|\{ required: string; type: \{ email: string; url: string; string: string; number: string; array: string; object: string; boolean: string; \}; number: \{ min: string; max: string; equal: string; range: string; positive: string; negative: string; \}; string: \{ \.\.\.; \}; array: \{ \.\.\.; \}; object: \{ \.\.\.; \}; boolean: \{ \.\.\.; \}; \}|
