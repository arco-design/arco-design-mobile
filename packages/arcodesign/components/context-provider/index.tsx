import React, { createContext, useState, useEffect, useCallback } from 'react';
import { addCssRules, removeCssStyleDom, defaultLocale, ILocale } from '@arco-design/mobile-utils';

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
    /**
     * 手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用
     * @en Manually control the current system, and the incoming value will be used directly after being passed in. It is applicable when the initial value of the system needs to be specified in the ssr scenario.
     * @default ""
     */
    system?: 'pc' | 'android' | 'ios' | '';
    /**
     * 是否使用暗黑模式
     * @en Whether to use dark mode
     * @default false
     */
    useDarkMode?: boolean;
    /**
     * 是否处于暗黑模式，指定后以指定的值为准
     * @en Whether it is in dark mode
     * @default false
     */
    isDarkMode?: boolean;
    /**
     * 主题变量，传入后将在线替换css变量，需设置less变量 @use-css-vars: 1
     * @en Theme variable. The css variable will be replaced online after input. The less variable needs to be set @use-css-vars: 1
     */
    theme?: Record<string, string>;
    /**
     * 国际化语言包配置
     * @en Internationalized language configuration
     */
    locale?: ILocale;
    /**
     * 是否使用Rtl模式
     * @en Whether to use rtl
     * @default false
     */
    useRtl?: boolean;
}

export const defaultContext: GlobalContextParams = {
    prefixCls: 'arco',
    system: '',
    useDarkMode: false,
    isDarkMode: false,
    locale: defaultLocale,
    useRtl: false,
};

export const GlobalContext = createContext<GlobalContextParams>(defaultContext);

export interface ContextProviderProps extends GlobalContextParams {
    children: React.ReactNode;
}

export type WithGlobalContext<T> = T & { context?: GlobalContextParams };

/**
 * 全局数据控制组件，用于替换全局数据。
 * @en Global data control component, used to replace global data.
 * @type 其他
 * @type_en Others
 * @name 全局配置
 * @name_en ContextProvider
 */
export default function ContextProvider(props: ContextProviderProps) {
    const {
        children,
        useDarkMode,
        isDarkMode,
        theme,
        locale = defaultLocale,
        ...restProps
    } = props;
    const [isDarkModeState, setIsDarkModeState] = useState<boolean>();
    const changeDarkMode = useCallback(
        (res: MediaQueryListEvent) => setIsDarkModeState(res.matches),
        [],
    );

    useEffect(() => {
        const matchMedia = window.matchMedia('(prefers-color-scheme: dark)');
        if (useDarkMode) {
            const dark = matchMedia.matches;
            setIsDarkModeState(dark);
            if (typeof matchMedia.addEventListener === 'function') {
                matchMedia.addEventListener('change', changeDarkMode);
            } else if (typeof matchMedia.addListener === 'function') {
                matchMedia.addListener(changeDarkMode);
            }
        }
        return () => {
            if (useDarkMode) {
                if (typeof matchMedia.removeEventListener === 'function') {
                    matchMedia.removeEventListener('change', changeDarkMode);
                } else if (typeof matchMedia.removeListener === 'function') {
                    matchMedia.removeListener(changeDarkMode);
                }
            }
        };
    }, [useDarkMode]);

    useEffect(() => {
        if (!theme) {
            return;
        }
        addCssRules('arcoCustomTheme', theme);
        return () => {
            removeCssStyleDom('arcoCustomTheme');
        };
    }, [theme]);

    return (
        <GlobalContext.Provider
            value={{
                ...defaultContext,
                ...restProps,
                isDarkMode: isDarkMode === void 0 ? isDarkModeState : isDarkMode,
                useDarkMode,
                locale: locale || defaultLocale,
            }}
        >
            {children}
        </GlobalContext.Provider>
    );
}

export const ContextLayout = GlobalContext.Consumer;

export function CompWithGlobalContext<P extends JSX.IntrinsicAttributes>(
    Component: React.FunctionComponent<P>,
) {
    return function (props: WithGlobalContext<P>) {
        const { context: propsContext, ...others } = props;
        return (
            <ContextLayout>
                {context => (
                    <ContextProvider {...{ ...context, ...propsContext }}>
                        <Component {...(others as P)} />
                    </ContextProvider>
                )}
            </ContextLayout>
        );
    };
}
