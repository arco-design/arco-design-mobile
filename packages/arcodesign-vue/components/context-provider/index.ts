import ContextProvider from './ContextProvider.vue';
import { installFactory } from '../_helpers/install';

export type * from './type';
export * from './utils';

const install = installFactory(ContextProvider);

export { ContextProvider };

export default {
    install,
};
