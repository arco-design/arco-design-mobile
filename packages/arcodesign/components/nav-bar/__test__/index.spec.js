import React from 'react';
import demoTest from '../../../tests/demoTest';
import mountTest from '../../../tests/mountTest';
import NavBar from '..';
import { mount } from 'enzyme';
import { defaultContext } from '../../context-provider';
import { act } from 'react-dom/test-utils';
import { mockAddListener, createStartTouchEventObject, createMoveTouchEventObject } from '../../../tests/helpers/mockEvent';
import { mockContainerSize, resetContainerSizeMock } from '../../../tests/helpers/mockElement';

const prefix = `${defaultContext.prefixCls}-nav-bar`;

demoTest('nav-bar');

mountTest(NavBar, 'NavBar');

describe('NavBar style', () => {
    it('Common styles render correctly', () => {
        const wrapper = mount(
            <NavBar
                fixed={false}
                title="title"
                rightContent={<span>more</span>}/>
        );
        expect(wrapper.find(`.${prefix}-left`).length).toBe(1);
        expect(wrapper.find(`.${prefix}-right`).length).toBe(1);
    });
})

describe('NavBar actions', () => {

    beforeAll(() => {
        mockContainerSize();
    });

    afterAll(() => {
        resetContainerSizeMock();
    });

    beforeEach(() => {
        jest.useFakeTimers();
    });

    afterEach(() => {
        jest.useRealTimers();
    });

    it('Click render correctly', () => {
        const onClickLeft = jest.fn();
        const onClickRight = jest.fn();
        const wrapper = mount(
            <NavBar
                title="title"
                rightContent={<span>more</span>}
                onClickLeft={onClickLeft}
                onClickRight={onClickRight}
            />
        );
        wrapper.find(`.${prefix}-left`).simulate('click');
        wrapper.find(`.${prefix}-right`).simulate('click');
        expect(onClickLeft.mock.calls.length).toBe(1);
        expect(onClickRight.mock.calls.length).toBe(1);
    });


    it('should callback correctly when scrolled', () => {
        const wrapper = mount(
            <NavBar
                statusBarHeight={10}
                showOffset={20}
                title="title"
                getComputedStyleByScroll={() => window}
            />
        );
        const map = mockAddListener(wrapper.find(`.${prefix}`));
        wrapper.setProps({ showOffset: 20 });
        act(() => {
            map.touchstart(createStartTouchEventObject({ x: 0, y: 0 }));
            map.touchmove(createMoveTouchEventObject({ x: 0, y: 100 }));
            jest.advanceTimersByTime(1000);
        });
        act(() => {
            map.touchend(createMoveTouchEventObject({ x: 0, y: 200 }));
        });
        wrapper.update();
    });
})
