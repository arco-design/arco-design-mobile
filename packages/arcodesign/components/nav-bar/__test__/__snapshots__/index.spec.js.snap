// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav-bar demo test nav-bar demo: index.md renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-label=""
    class="arco-nav-bar"
    role="banner"
    style="padding-top: 0px;"
  >
    <div
      class="pc arco-nav-bar-wrapper"
      style="padding-top: 0px;"
    >
      <div
        class="arco-nav-bar-inner all-border-box"
      >
        <div
          class="arco-nav-bar-left"
        >
          <div
            aria-label="返回"
            class="c-svg-arrow nav-bar-back"
            role="button"
          >
            <svg
              height="100%"
              viewBox="0 0 16 16"
            >
              <path
                d="M2.1,8l5.4,5.4c0.1,0.1,0.1,0.3,0,0.5L7,14.4c-0.1,0.1-0.3,0.1-0.5,0L0.7,8.5c-0.3-0.3-0.3-0.7,0-0.9 l5.9-5.9c0.1-0.1,0.3-0.1,0.5,0l0.5,0.5c0.1,0.1,0.1,0.3,0,0.5L2.1,8z"
                fill="currentColor"
                id="path-1_1_"
              />
            </svg>
          </div>
        </div>
        <div
          class="arco-nav-bar-title"
        >
          <div
            class="arco-nav-bar-title-text"
          >
            Title
          </div>
        </div>
        <div
          class="arco-nav-bar-right"
        >
          <span
            style="color: rgb(22, 93, 255);"
          >
            More
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="gap-line"
  />
  <div
    aria-label=""
    class="arco-nav-bar"
    role="banner"
    style="padding-top: 0px;"
  >
    <div
      class="pc arco-nav-bar-wrapper"
      style="padding-top: 0px;"
    >
      <div
        class="arco-nav-bar-inner all-border-box"
      >
        <div
          class="arco-nav-bar-left"
        >
          <div
            aria-label="返回"
            class="c-svg-arrow nav-bar-back"
            role="button"
          >
            <svg
              height="100%"
              viewBox="0 0 16 16"
            >
              <path
                d="M2.1,8l5.4,5.4c0.1,0.1,0.1,0.3,0,0.5L7,14.4c-0.1,0.1-0.3,0.1-0.5,0L0.7,8.5c-0.3-0.3-0.3-0.7,0-0.9 l5.9-5.9c0.1-0.1,0.3-0.1,0.5,0l0.5,0.5c0.1,0.1,0.1,0.3,0,0.5L2.1,8z"
                fill="currentColor"
                id="path-1_1_"
              />
            </svg>
          </div>
        </div>
        <div
          class="arco-nav-bar-title"
        >
          <div
            class="arco-nav-bar-title-text"
          >
            Title
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="gap-line"
  />
  <div
    aria-label=""
    class="arco-nav-bar"
    role="banner"
    style="padding-top: 0px; color: white; background: rgb(22, 93, 255);"
  >
    <div
      class="pc arco-nav-bar-wrapper"
      style="padding-top: 0px;"
    >
      <div
        class="arco-nav-bar-inner all-border-box"
      >
        <div
          class="arco-nav-bar-left"
        >
          <div
            aria-label="返回"
            class="c-svg-arrow nav-bar-back"
            role="button"
          >
            <svg
              height="100%"
              viewBox="0 0 16 16"
            >
              <path
                d="M2.1,8l5.4,5.4c0.1,0.1,0.1,0.3,0,0.5L7,14.4c-0.1,0.1-0.3,0.1-0.5,0L0.7,8.5c-0.3-0.3-0.3-0.7,0-0.9 l5.9-5.9c0.1-0.1,0.3-0.1,0.5,0l0.5,0.5c0.1,0.1,0.1,0.3,0,0.5L2.1,8z"
                fill="currentColor"
                id="path-1_1_"
              />
            </svg>
          </div>
        </div>
        <div
          class="arco-nav-bar-title"
        >
          <div
            class="arco-nav-bar-title-text"
          >
            Message
          </div>
        </div>
        <div
          class="arco-nav-bar-right"
        >
          <svg
            class="arco-icon arco-icon-more "
            fill="currentColor"
            height="1em"
            viewBox="0 0 48 48"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M35 21h6v6h-6zM7 21h6v6H7zM21 21h6v6h-6z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`nav-bar demo test nav-bar demo: index2.md renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-label=""
    class="arco-nav-bar"
    role="banner"
    style="padding-top: 0px;"
  >
    <div
      class="pc arco-nav-bar-wrapper"
      style="padding-top: 0px;"
    >
      <div
        class="arco-nav-bar-inner all-border-box"
      >
        <div
          class="arco-nav-bar-title"
        >
          <div
            class="arco-nav-bar-title-text"
          >
            Title
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`nav-bar demo test nav-bar demo: index3.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="gap-line"
  />
  <div
    aria-label=""
    class="arco-nav-bar"
    role="banner"
    style="padding-top: 0px;"
  >
    <div
      class="pc arco-nav-bar-wrapper"
      style="padding-top: 0px;"
    >
      <div
        class="arco-nav-bar-inner all-border-box"
      >
        <div
          class="arco-nav-bar-left"
        >
          <div
            aria-label="返回"
            class="c-svg-arrow nav-bar-back"
            role="button"
          >
            <svg
              height="100%"
              viewBox="0 0 16 16"
            >
              <path
                d="M2.1,8l5.4,5.4c0.1,0.1,0.1,0.3,0,0.5L7,14.4c-0.1,0.1-0.3,0.1-0.5,0L0.7,8.5c-0.3-0.3-0.3-0.7,0-0.9 l5.9-5.9c0.1-0.1,0.3-0.1,0.5,0l0.5,0.5c0.1,0.1,0.1,0.3,0,0.5L2.1,8z"
                fill="currentColor"
                id="path-1_1_"
              />
            </svg>
          </div>
        </div>
        <div
          class="arco-nav-bar-title"
        >
          <div
            class="arco-nav-bar-title-text"
          >
            Title
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="gap-line"
  />
  <div
    aria-label=""
    class="arco-nav-bar"
    role="banner"
    style="padding-top: 0px;"
  >
    <div
      class="pc arco-nav-bar-wrapper"
      style="padding-top: 0px;"
    >
      <div
        class="arco-nav-bar-inner all-border-box"
      >
        <div
          class="arco-nav-bar-left"
        >
          <div
            aria-label="返回"
            class="c-svg-arrow nav-bar-back"
            role="button"
          >
            <svg
              height="100%"
              viewBox="0 0 16 16"
            >
              <path
                d="M2.1,8l5.4,5.4c0.1,0.1,0.1,0.3,0,0.5L7,14.4c-0.1,0.1-0.3,0.1-0.5,0L0.7,8.5c-0.3-0.3-0.3-0.7,0-0.9 l5.9-5.9c0.1-0.1,0.3-0.1,0.5,0l0.5,0.5c0.1,0.1,0.1,0.3,0,0.5L2.1,8z"
                fill="currentColor"
                id="path-1_1_"
              />
            </svg>
          </div>
        </div>
        <div
          class="arco-nav-bar-title"
        >
          <div
            class="arco-nav-bar-title-text"
          >
            Title
          </div>
        </div>
        <div
          class="arco-nav-bar-right"
        >
          <div
            style="display: flex; align-items: center;"
          >
            <span
              style="margin-right: 16px;"
            >
              Edit
            </span>
            <svg
              class="arco-icon arco-icon-more "
              fill="currentColor"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M35 21h6v6h-6zM7 21h6v6H7zM21 21h6v6h-6z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="gap-line"
  />
  <div
    aria-label=""
    class="arco-nav-bar"
    role="banner"
    style="padding-top: 0px; background: url(https://sf1-cdn-tos.toutiaostatic.com/obj/arco-mobile/_static_/navbar-demo-bg.png); color: white;"
  >
    <div
      class="immersive-navbar pc arco-nav-bar-wrapper arco-nav-bar-wrapper-border"
      style="padding-top: 10px;"
    >
      <div
        class="arco-nav-bar-inner all-border-box"
      >
        <div
          class="arco-nav-bar-left"
        >
          <div
            aria-label="返回"
            class="c-svg-arrow nav-bar-back"
            role="button"
          >
            <svg
              height="100%"
              viewBox="0 0 16 16"
            >
              <path
                d="M2.1,8l5.4,5.4c0.1,0.1,0.1,0.3,0,0.5L7,14.4c-0.1,0.1-0.3,0.1-0.5,0L0.7,8.5c-0.3-0.3-0.3-0.7,0-0.9 l5.9-5.9c0.1-0.1,0.3-0.1,0.5,0l0.5,0.5c0.1,0.1,0.1,0.3,0,0.5L2.1,8z"
                fill="currentColor"
                id="path-1_1_"
              />
            </svg>
          </div>
        </div>
        <div
          class="arco-nav-bar-title"
        >
          <div
            class="arco-nav-bar-title-text"
          >
            I am Immersive Navbar Title
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`nav-bar demo test nav-bar demo: index4.md renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-label=""
    class="arco-nav-bar"
    role="banner"
    style="padding-top: 0px;"
  >
    <div
      class="pc arco-nav-bar-wrapper"
      style="padding-top: 0px;"
    >
      <div
        class="arco-nav-bar-inner all-border-box"
      >
        <div
          class="arco-nav-bar-left"
        >
          <svg
            class="arco-icon arco-icon-close "
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M509.568 447.083L796.16 160.469a21.333 21.333 0 0130.165 0l30.166 30.166a21.333 21.333 0 010 30.186L569.899 507.435l286.613 286.592a21.333 21.333 0 010 30.165l-30.165 30.165a21.333 21.333 0 01-30.166 0L509.568 567.765 222.933 854.38a21.333 21.333 0 01-30.186 0l-30.166-30.166a21.333 21.333 0 010-30.165l286.614-286.613L162.603 220.8a21.333 21.333 0 010-30.187l30.165-30.165a21.333 21.333 0 0130.187 0l286.613 286.613z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="arco-nav-bar-title"
        >
          <div
            class="arco-nav-bar-title-text"
          >
            Title
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="gap-line"
  />
  <div
    aria-label=""
    class="arco-nav-bar"
    role="banner"
    style="padding-top: 0px;"
  >
    <div
      class="pc arco-nav-bar-wrapper"
      style="padding-top: 0px;"
    >
      <div
        class="arco-nav-bar-inner all-border-box"
      >
        <div
          class="arco-nav-bar-left"
        >
          <div
            aria-label="返回"
            class="c-svg-arrow nav-bar-back"
            role="button"
          >
            <svg
              height="100%"
              viewBox="0 0 16 16"
            >
              <path
                d="M2.1,8l5.4,5.4c0.1,0.1,0.1,0.3,0,0.5L7,14.4c-0.1,0.1-0.3,0.1-0.5,0L0.7,8.5c-0.3-0.3-0.3-0.7,0-0.9 l5.9-5.9c0.1-0.1,0.3-0.1,0.5,0l0.5,0.5c0.1,0.1,0.1,0.3,0,0.5L2.1,8z"
                fill="currentColor"
                id="path-1_1_"
              />
            </svg>
          </div>
        </div>
        <div
          class="arco-nav-bar-title"
        >
          <div
            class="arco-nav-bar-title-text"
          >
            Title
          </div>
        </div>
        <div
          class="arco-nav-bar-right"
        >
          <svg
            class="arco-icon arco-icon-search "
            height="1em"
            style="font-size: 18px;"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M469.333 106.667C669.632 106.667 832 269.035 832 469.333c0 75.712-23.19 146.006-62.87 204.16l144.683 144.683a21.333 21.333 0 010 30.165l-36.202 36.203a21.333 21.333 0 01-30.166 0L706.56 743.68A361.259 361.259 0 01469.333 832c-200.298 0-362.666-162.368-362.666-362.667s162.368-362.666 362.666-362.666zm0 89.6c-150.826 0-273.066 122.24-273.066 273.066S318.507 742.4 469.333 742.4 742.4 620.16 742.4 469.333 620.16 196.267 469.333 196.267z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`nav-bar demo test nav-bar demo: index5.md renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-label=""
    class="arco-nav-bar"
    role="banner"
    style="padding-top: 0px;"
  >
    <div
      class="pc arco-nav-bar-wrapper"
      style="padding-top: 0px;"
    >
      <div
        class="arco-nav-bar-inner all-border-box"
      >
        <div
          class="arco-nav-bar-left"
        >
          <div
            aria-label="返回"
            class="c-svg-arrow nav-bar-back"
            role="button"
          >
            <svg
              height="100%"
              viewBox="0 0 16 16"
            >
              <path
                d="M2.1,8l5.4,5.4c0.1,0.1,0.1,0.3,0,0.5L7,14.4c-0.1,0.1-0.3,0.1-0.5,0L0.7,8.5c-0.3-0.3-0.3-0.7,0-0.9 l5.9-5.9c0.1-0.1,0.3-0.1,0.5,0l0.5,0.5c0.1,0.1,0.1,0.3,0,0.5L2.1,8z"
                fill="currentColor"
                id="path-1_1_"
              />
            </svg>
          </div>
        </div>
        <div
          style="width: 100%; padding: 1px 46px;"
        >
          <div
            class="arco-tabs arco-tabs-top arco-tabs-vertical all-border-box"
          >
            <div
              class="arco-tab-cell-container-wrap vertical type-line-divide pc"
            >
              <div
                class="arco-tab-cell-container-inner vertical pos-top type-line-divide pc"
              >
                <div
                  class="arco-tab-cell-container vertical pos-top arrange-center type-line-divide shown"
                >
                  <div
                    class="arco-tab-cell vertical line-divide pc active no-shrink"
                  >
                    Option 1
                  </div>
                  <div
                    class="arco-tab-cell vertical line-divide pc no-shrink"
                  >
                    Option 2
                  </div>
                  <div
                    class="arco-tab-cell vertical line-divide pc no-shrink last"
                  >
                    Option 3
                  </div>
                  <div
                    class="arco-tab-cell-underline"
                    style="transition-duration: 0ms; transform: translateX(0px) translateZ(0);"
                  >
                    <div
                      class="arco-tab-cell-underline-inner vertical"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="arco-tab-pane-container vertical mode-swipe"
              style="width: 0%; transform: translateX(0px) translateZ(0); transition-duration: 0ms;"
            />
          </div>
        </div>
        <div
          class="arco-nav-bar-right"
        >
          <svg
            class="arco-icon arco-icon-more "
            fill="currentColor"
            height="1em"
            viewBox="0 0 48 48"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M35 21h6v6h-6zM7 21h6v6H7zM21 21h6v6h-6z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`nav-bar demo test nav-bar demo: index6.md renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-label=""
    class="arco-nav-bar"
    role="banner"
    style="padding-top: 0px;"
  >
    <div
      class="pc arco-nav-bar-wrapper"
      style="padding-top: 0px;"
    >
      <div
        class="arco-nav-bar-inner all-border-box"
      >
        <div
          class="arco-nav-bar-left"
        >
          Cancel
        </div>
        <div
          class="arco-nav-bar-title"
        >
          <div
            class="arco-nav-bar-title-text"
          >
            Message
          </div>
        </div>
        <div
          class="arco-nav-bar-right"
        >
          <span
            style="background: rgb(22, 93, 255); border-radius: 26px; width: 60px; height: 28px; display: flex; align-items: center; justify-content: center; font-size: 14px; color: rgb(255, 255, 255);"
          >
            Finish
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="gap-line"
  />
</DocumentFragment>
`;
